courses/utils.py

```diff
@@ -126,3 +129,54 @@ def generate_program_certificate(user, program):
         )
 
     return program_cert, True
+
+
+def sync_course_runs(runs):
+    """
+    Sync course run dates and title from Open edX
+
+    Args:
+        runs ([CourseRun]): list of CourseRun objects.
+
+    Returns:
+        [str], [str]: Lists of success and error logs respectively
+    """
+    api_client = get_edx_api_course_detail_client()
+
+    success_logs = []
+    error_logs = []
+
+    # Iterate all eligible runs and sync if possible
+    for run in runs:
+        try:
+            course_detail = api_client.get_detail(run.courseware_id)
+
+            # Reset the expiration_date so it is calculated automatically and
+            # does not raise a validation error now that the start or end date
+            # has changed.
+            if (
+                run.start_date != course_detail.start
+                or run.end_date != course_detail.end
+            ):
+                run.expiration_date = None
+
+            run.title = course_detail.name
+            run.start_date = course_detail.start
+            run.end_date = course_detail.end
+            run.enrollment_start = course_detail.enrollment_start
+            run.enrollment_end = course_detail.enrollment_end
+            try:
+                run.save()
```

Comment:
do we need to wrap it under `with  transaction.atomic():` call ?

Reply:
That would mean either all runs are updated or none. What's your thoughts on this @pdpinch ? Do you want to allow partial updates or not?

Reply:
Atomic for each course would be OK. But it shouldn't be atomic for the whole update. 

Please make sure to raise any exceptions to Sentry.

Reply:
Done